packages/controller-utils/src/siwe.ts

```diff
@@ -34,10 +34,130 @@ function msgHexToText(hex: string): string {
   }
 }
 
+/**
+ * @type WrappedSIWERequest
+ *
+ * Sign-In With Ethereum (SIWE)(EIP-4361) message with request metadata
+ * @property {string} from - Subject account address
+ * @property {string} origin - The RFC 3986 originating authority of the signing request, including scheme
+ * @property {ParsedMessage} siwe - The data parsed from the message
+ */
+export interface WrappedSIWERequest {
+  from: string;
+  origin: string;
+  siwe: SIWEMessage;
+}
+
+interface HostParts extends DomainParts {
+  scheme: string;
+}
+
+interface DomainParts {
+  userinfo: string | null;
+  host: string;
+  port: number | null;
+}
+
+export const ORIGIN_REGEX =
+  /^(?<scheme>[^:/]+):\/\/((?<userinfo>[^@]*)@)?(?<host>[^:@/?]+)(:(?<port>[0-9]+))?/u;
+export const DOMAIN_REGEX =
+  /^((?<userinfo>[^@]*)@)?(?<host>[^:@/?]+)(:(?<port>[0-9]+))?$/u;
+const DEFAULT_PORTS_BY_SCHEME = {
+  http: 80,
+  https: 443,
+} as { [key: string]: number | undefined };
+
+/**
+ * Parses parts from RFC 3986 authority from environment relevant for EIP-4361 origin validation.
+ *
+ * @param authority - input string
+ * @returns parsed parts
+ */
+export const parseAuthorityParts = (authority: string): HostParts => {
```

Comment:
I think we can update the parameter to indicate we are passing the origin rather than the authority. I think the authority is part of the origin and does not include the scheme

--- 

maybe rename the function to `parseOriginParts' or rename `parseDomainParts` to `parseDomainAuthorityParts`.